docs/dev/data_validation.rst

```diff
@@ -0,0 +1,688 @@
+  are somewhat analogous to `Dagster assets
+  <https://docs.dagster.io/guides/build/assets/defining-assets>`_.
+* As a SQL-based tool, dbt generally expects to be querying a database. However, in our
+  case the tables are stored as Apache Parquet files, which we query with SQL via
+  DuckDB. This means some of dbt's functionality is not available. For example, we can't
+  use `the dbt adapter object
+  <https://docs.getdbt.com/reference/dbt-jinja-functions/adapter>`_ in our test
+  definitions because it relies on being able to access the underlying database schema,
+  which doesn't exist (at least, not within the dbt context).
+
+To run all of the data validation tests, from within the ``pudl/dbt/`` directory run:
+
+.. code-block:: bash
+
+   dbt build
+
+For more fine-grained control, you can use the ``--select`` option to run only the tests
+defined for a particular table, or all instances of a particular test no matter what
+table it's associated with. Or you can combine the two to run just a particular test
+on a particular table. Some examples:
+
+.. code-block:: bash
+
+   # Run all tests defined for the out_eia__monthly_generators table
+   dbt build --select "source:pudl.out_eia__monthly_generators"
+   # Run all instances of the expect_columns_not_all_null test
+   dbt build --select "test_name:expect_columns_not_all_null"
+   # Run expect_columns_not_all_null test on the out_eia__monthly_generators table only
+   dbt build --select "test_name:expect_columns_not_all_null,source:pudl.out_eia__monthly_generators"
+   # Use a wildcard "*" to run all tests on tables whose names start with out_eia923__
+   dbt build --select "source:pudl.out_eia923__*"
+
+Similarly, you can exclude individual tables or tests using ``--exclude``. One case
+where this is useful is running the data validation tests against the outputs of the
+fast ETL. We do not store expected row-counts for the fast ETL outputs, and so generally
+expect the row-count checks to fail. To run all of the data validation tests except for
+the row counts and avoid seeing all those spurious failures you could run:
+
+.. code-block:: bash
+
+   dbt build --exclude "test_name:check_row_counts_per_partition"
+
+For more options, see the `dbt selection syntax documentation
+<https://docs.getdbt.com/reference/node-selection/syntax>`_.
+
+.. note::
+
+   The dbt tests can be run in parallel to speed them up with the ``--threads`` argument
+   but this sometimes results in spurious errors like "too many files open" which are
+   not related to the data being tested.
+
+.. note::
+
+   There are a handful of data validation tests that have been implemented using
+   `Dagster's asset checks <https://docs.dagster.io/guides/test/asset-checks>`_.
+   Typically these tests weren't well suited to SQL, weren't performance bottlenecks,
+   and had already been implemented in Python. E.g. :func:`pudl.validate.no_null_rows`.
+
+The ``dbt_helper`` script
+~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Rather than trying to repeat all of the `dbt_helper` documentation here and make it a standalone section, I felt like it was more natural to work it into each of the thematic sections where it had something to offer. I think having the complete, canonical documentation for it be in the module / help messages makes sense.